From 5740afa79c8d10e92c8050df77c19b4ddd710491 Mon Sep 17 00:00:00 2001
Message-Id: <5740afa79c8d10e92c8050df77c19b4ddd710491.1681233514.git.vinu.vaghasia@amd.com>
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Tue, 11 Apr 2023 12:18:07 -0500
Subject: [PATCH] u-boot-aspped: Modify u-boot code to support G304 platform
 devices

- Hawaii dts modified to disapbe qspi and reduce frequency
- GPIOV4 and GPIOI5 enabled by setting SCU51C[10]=0
- JEDEC code added for is25wp256

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/dts/ast2600-hawaii.dts | 31 +++++++++++++++++++------------
 common/main.c                   | 12 ++++++++++++
 drivers/mtd/spi/spi-nor-ids.c   |  3 +++
 3 files changed, 34 insertions(+), 12 deletions(-)

diff --git a/arch/arm/dts/ast2600-hawaii.dts b/arch/arm/dts/ast2600-hawaii.dts
index 7db50325c8..0ac05e26e6 100644
--- a/arch/arm/dts/ast2600-hawaii.dts
+++ b/arch/arm/dts/ast2600-hawaii.dts
@@ -100,33 +100,40 @@

 // FPGA and BIOS FLASH

+#if 0
 &spi1 {
 	status = "okay";
-
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_spi1_default &pinctrl_spi1quad_default>;
+	pinctrl-0 = <&pinctrl_spi1_default >;

 	flash@0 {
-		compatible = "spi-flash";
 		status = "okay";
-		spi-max-frequency = <50000000>;
-		spi-tx-bus-width = <4>;
-		spi-rx-bus-width = <4>;
+		spi-max-frequency = <500000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
+	};
+
+	flash@1 {
+		status = "okay";
+		spi-max-frequency = <500000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
 	};
+
+
 };
+#endif

 &spi2 {
 	status = "okay";
-
 	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_spi2_default &pinctrl_spi2quad_default>;
+	pinctrl-0 = <&pinctrl_spi2_default>;

 	flash@0 {
-		compatible = "spi-flash";
 		status = "okay";
-		spi-max-frequency = <50000000>;
-		spi-tx-bus-width = <4>;
-		spi-rx-bus-width = <4>;
+		spi-max-frequency = <10000000>;
+		spi-tx-bus-width = <1>;
+		spi-rx-bus-width = <1>;
 	};
 };

diff --git a/common/main.c b/common/main.c
index caf2dacaa1..7e4d72cec0 100644
--- a/common/main.c
+++ b/common/main.c
@@ -26,6 +26,9 @@
 #define SIX_DIMM_PER_BUS   6
 #define TMP_BUF_LEN        8

+#define ASPEED_SYS_SCU_HW_STRAP3 0x1e6e251C  // Hardware Strap3
+#define HW_STRAP3_MASK           0xFFFFFBFF  // Bit# 10
+
 /*
  * Board-specific Platform code can reimplement show_boot_progress () if needed
  */
@@ -252,6 +255,8 @@ static void run_preboot_environment_command(void)
 void main_loop(void)
 {
 	const char *s;
+	u32 scu_val;
+
 	u32 por_rst = readl(ASPEED_SYS_SCRATCH_1FC);

 	bootstage_mark_name(BOOTSTAGE_ID_MAIN_LOOP, "main_loop");
@@ -277,6 +282,13 @@ void main_loop(void)
 	/* Read board id from eerpom and set env */
 	set_board_id();
 	set_hostname();
+
+	/* Set Hardware Strap3 reg */
+	scu_val = readl(ASPEED_SYS_SCU_HW_STRAP3);
+	scu_val = scu_val & (u32)HW_STRAP3_MASK;
+	writel(scu_val, ASPEED_SYS_SCU_HW_STRAP3);
+	printf("HWStrap3 = 0x%08x\n",scu_val);
+
 	s = bootdelay_process();
 	if (cli_process_fdt(&s))
 		cli_secure_boot_cmd(s);
diff --git a/drivers/mtd/spi/spi-nor-ids.c b/drivers/mtd/spi/spi-nor-ids.c
index 5882eab2e3..b824a36aa4 100644
--- a/drivers/mtd/spi/spi-nor-ids.c
+++ b/drivers/mtd/spi/spi-nor-ids.c
@@ -152,6 +152,9 @@ const struct flash_info spi_nor_ids[] = {
 			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
 	{ INFO("is25wp128",  0x9d7018, 0, 64 * 1024, 256,
 			SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ) },
+	{ INFO("is25wp256", 0x9d7019, 0, 64 * 1024, 512,
+            SECT_4K | SPI_NOR_DUAL_READ | SPI_NOR_QUAD_READ | SPI_NOR_4B_OPCODES) },
+
 #endif
 #ifdef CONFIG_SPI_FLASH_MACRONIX	/* MACRONIX */
 	/* Macronix */
--
2.17.1

